src/lib/stepper/stepper.spec.ts

```diff
@@ -0,0 +1,559 @@
+    TestBed.compileComponents();
+  }));
+
+  describe('basic horizontal stepper', () => {
+    let fixture: ComponentFixture<SimpleMdHorizontalStepperApp>;
+    let stepperComponent: MdHorizontalStepper;
+    let stepperEl: HTMLElement;
+
+    beforeEach(() => {
+      fixture = TestBed.createComponent(SimpleMdHorizontalStepperApp);
+      fixture.detectChanges();
+
+      stepperComponent = fixture.debugElement
+          .query(By.css('md-horizontal-stepper')).componentInstance;
+    });
+
+    it('should default to the first step', () => {
+      expect(stepperComponent.selectedIndex).toBe(0);
+    });
+
+    it('should change selected index on header click', () => {
+      let stepHeader = fixture.debugElement.queryAll(By.css('.mat-horizontal-stepper-header'));
+      checkSelectionChangeOnHeaderClick(stepperComponent, fixture, stepHeader);
+
+    });
+
+    it('should set the "tablist" role on stepper', () => {
+      stepperEl = fixture.debugElement.query(By.css('md-horizontal-stepper')).nativeElement;
+      expect(stepperEl.getAttribute('role')).toBe('tablist');
+    });
+
+    it('should expand the right content', () => {
+      let stepContent = fixture.debugElement.queryAll(By.css(`.mat-horizontal-stepper-content`));
+      checkExpandedContent(stepperComponent, fixture, stepContent);
+    });
+
+    it('should display the correct label', () => {
+      checkCorrectLabel(stepperComponent, fixture);
+    });
+
+    it('should go to next available step when the next button is clicked', () => {
+      checkNextStepperButton(stepperComponent, fixture);
+    });
+
+    it('should go to previous available step when the previous button is clicked', () => {
+      checkPreviousStepperButton(stepperComponent, fixture);
+    });
+
+    it('should set the correct step position for animation', () => {
+      checkStepPosition(stepperComponent, fixture);
+    });
+
+    it('should support keyboard events to move and select focus', () => {
+      let stepHeader = fixture.debugElement.queryAll(By.css('.mat-horizontal-stepper-header'));
+      checkKeyboardEvent(stepperComponent, fixture, stepHeader);
+    });
+  });
+
+  describe('linear horizontal stepper', () => {
+    let fixture: ComponentFixture<LinearMdHorizontalStepperApp>;
+    let testComponent: LinearMdHorizontalStepperApp;
+    let stepperComponent: MdHorizontalStepper;
+
+    beforeEach(() => {
+      fixture = TestBed.createComponent(LinearMdHorizontalStepperApp);
+      fixture.detectChanges();
+
+      testComponent = fixture.componentInstance;
+      stepperComponent = fixture.debugElement
+          .query(By.css('md-horizontal-stepper')).componentInstance;
+    });
+
+    it('should have true linear attribute', () => {
+      expect(stepperComponent.linear).toBe(true);
+    });
+
+    it('should not move to next step if current step is not valid', () => {
+      expect(testComponent.oneGroup.get('oneCtrl')!.value).toBe('');
+      expect(testComponent.oneGroup.valid).toBe(false);
```

Comment:
Nit: Here I'd expect a test that the control itself is invalid (if you were to add another control, it could cause the parent form to be false for a different reason).  Same on 199.